dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/FunctionChoiceBehavior.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Represents the base class for different function choice behaviors.
+/// </summary>
+[JsonPolymorphic(TypeDiscriminatorPropertyName = "type")]
+[JsonDerivedType(typeof(AutoFunctionChoiceBehavior), typeDiscriminator: AutoFunctionChoiceBehavior.TypeDiscriminator)]
+[JsonDerivedType(typeof(RequiredFunctionChoiceBehavior), typeDiscriminator: RequiredFunctionChoiceBehavior.TypeDiscriminator)]
+[JsonDerivedType(typeof(NoneFunctionChoiceBehavior), typeDiscriminator: NoneFunctionChoiceBehavior.TypeDiscriminator)]
+[Experimental("SKEXP0001")]
+public abstract class FunctionChoiceBehavior
+{
+    /// <summary>The separator used to separate plugin name and function name.</summary>
+    protected const string FunctionNameSeparator = ".";
```

Comment:
I think this is currently a `-`, why is this being changed?

Reply:
The "." character seems to be used by all SK prompt templates as a separator between the plugin name and function name. So, in this case, it is used for consistency within the same prompt - the same separator for functions inside the prompt template and function choice behavior. To avoid confusion when the `Functions` collection is used in the code, we can either exclude it from the public surface, allowing functions to be supplied only via the constructor but allowing its deserialization, or have a converter that changes "." to "-" separators during deserialization.

Reply:
Fixed